internal/monitors/kubernetes/cluster/metrics/pods.go

```diff
@@ -11,7 +11,7 @@ import (
 	v1 "k8s.io/api/core/v1"
 )
 
-func datapointsForPod(pod *v1.Pod) []*datapoint.Datapoint {
+func datapointsForPod(pod *v1.Pod) ([]*datapoint.Datapoint, []*atypes.DimProperties) {
```

Comment:
This function should only return datapoints.  Why can't the dimProps stuff be done in the `dimPropsForPods` function below?

Reply:
`dimPropsForPods` does not have context about the pod object itself. It makes use of what we cache. I've moved this logic out to `handleAddPod` now.